rxbindings/src/main/java/com/amplifyframework/rx/RxStorageBinding.java

```diff
@@ -113,4 +116,25 @@
     private <T> Single<T> toSingle(RxAdapters.CancelableResultEmitter<T, StorageException> method) {
         return RxAdapters.toSingle(method);
     }
+
+    /**
+     * Defines the parameters of the download operation by
+     * supplying the generic types required by {@link RxAdapters.RxProgressAwareSingle}.
+     */
+    static final class RxStorageDownloadOperation extends
+        RxAdapters.RxProgressAwareSingle<StorageDownloadFileResult, StorageTransferProgress> {
+        RxStorageDownloadOperation(RxStorageDownloadCallbackMapper callbacks) {
+            super(callbacks);
+        }
+    }
+
+    /**
+     * Type alias that defines the generic parameters for a download operation.
+     * @see RxAdapters.RxProgressAwareCallbackMapper
+     */
+    interface RxStorageDownloadCallbackMapper
+        extends RxAdapters.RxProgressAwareCallbackMapper<StorageDownloadFileResult,
```

Comment:
If we only have this one thing that's progress aware, should we flatten these two interfaces into just one?

Reply:
Oh yeah...After moving some of this stuff around, I was able to reduce the footprint quite a bit.